codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -312,6 +315,14 @@ private void increaseInitialConnectionWindow(int deltaBytes) throws Http2Excepti
 
     final boolean consumeBytes(int streamId, int bytes) throws Http2Exception {
         Http2Stream stream = connection().stream(streamId);
+        // upgraded requests are ineligible for stream control
+        if (streamId == Http2CodecUtil.HTTP_UPGRADE_STREAM_ID) {
+            Boolean upgraded = stream.getProperty(upgradeKey);
+            if (Boolean.TRUE.equals(upgraded)) {
+                return true;
```

Comment:
continuation of https://github.com/netty/netty/pull/7400#discussion_r152084904 ...

This still doesn't make sense to me. Seems like if property is not null we should just return false, because we are not writing a window update and do not require a flush as a result of this method call.

Reply:
whoops, I just forgot to change it.